### ISSUE TITLE
Use bc math

### ISSUE THREAD

**Issue by mathewrapid** [triplepoint/php-units-of-measure#43]
http://php.net/manual/en/book.bc.php

**Comment by triplepoint** [comment]
Yeah, that's probably a good idea. I'll want to do some profiling to make sure we're not slowing things down too much, but I think this is probably an important improvement.

**Comment by HipsterJazzbo** [comment]
@triplepoint I've done some benchmarks out of curiosity, and for fewer than ~1,000 operations it seems to not really make a difference — either will take < 1ms. BCMath is a lot slower if you're wanting to do tens of thousands of operations, but we're talking ~20ms vs ~2ms, which seems reasonable to me within the intended purpose of this library.

There is also the issue that BCMath is not included by default with PHP (except on windows for some reason), but I think it seems reasonable to require it in order to be able to get accurate results out and avoid floating-point errors.

If that sounds reasonable to you, I'd be happy to go through and change everything to BCMath.

**Comment by soyuka** [comment]
What would it take to enable BCMath if available and the default if not?

**Comment by TomA-R** [comment]
hey @triplepoint this would be really useful, is this on the roadmap?

**Comment by soleson** [comment]
I've created a patch that can be installed via composer-patches. The patch defaults the precision to the value set by php. This was a quick-and-dirty fix, so I'm hoping others can review and improve the efficiency.
[php-unit-of-measure-bcmath-support.patch](https://github.com/PhpUnitsOfMeasure/php-units-of-measure/files/10468166/php-unit-of-measure-bcmath-support.patch)
